keps/sig-api-machinery/1965-kube-apiserver-identity/README.md

```diff
@@ -154,64 +198,135 @@ Alpha should provide basic functionality covered with tests described above.
+Old leases will be actively garbage collected by kube-controller-manager.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Stale Lease objects will remain stale (renewTime won't get updated)
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Stale Lease objects will be garbage collected.
+
+###### Are there any tests for feature enablement/disablement?
+
+There are some tests that require enabling the feature gate in [apiserver_identity_test.go](https://github.com/kubernetes/kubernetes/blob/24238425492227fdbb55c687fd4e94c8b58c1ee3/test/integration/controlplane/apiserver_identity_test.go).
+However, there are no tests validating feature enablement/disablement based on the gate. These tests should be added prior to Beta.
 
 ### Rollout, Upgrade and Rollback Planning
 
-_This section must be completed when targeting beta graduation to a release._
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Existing workloads should not be impacteded by this feature, unless they were
+looking for Lease objects in the `kube-apiserver-lease` namespace.
+
+###### What specific metrics should inform a rollback?
+
+Recently added [healthcheck metrics for apiserver](https://github.com/kubernetes/kubernetes/pull/112741), which includes
+the health of the post start hook can be used to inform rollback, specifically `kubernetes_healthcheck{poststarthook/start-kube-apiserver-identity-lease-controller}`
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Manual testing for upgrade/rollback will be done prior to Beta. Steps taken for manual tests will be updated here.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
 
 ### Monitoring Requirements
 
-_This section must be completed when targeting beta graduation to a release._
+###### How can an operator determine if the feature is in use by workloads?
+
+The existence of the `kube-apiserver-lease` namespace and Lease objects in the namespace
+will determine if the feature is working. Operators can check for clients that are accessing
+the Lease object to see if workloads or other controllers are relying on this feature.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [ ] Events
+  - Event Reason:
+- [X] API .status
+  - Condition name:
+  - Other field: `.spec.holderIdentity`, `.spec.acquireTime`, `.spec.renewTime`, `.spec.leaseTransitions`
+- [X] Other (treat as last resort)
+  - Details: audit logs for clients that are reading the Lease objects
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+A rough SLO here is that healthy kube-apiservers has a lease which is not older than 2 times the frequency of
```

Comment:
how would a cluster-admin know whether this has been achieved?  Perhaps its more practical to know that the number of leases in kube-apiserver-lease equals the number of kube-apiservers expected 95% of the time.

Reply:
> how would a cluster-admin know whether this has been achieved?

They could check the Lease's `spec.renewTime`, but I guess there would be no way of knowing if it was under some threshold 95% of the time. Let me add this as another potential SLO